src/lib/plugin_apis/fs.api

```diff
@@ -1188,4 +1269,84 @@ gboolean bd_fs_ntfs_resize (const gchar *device, guint64 new_size, GError **erro
  */
 BDFSNtfsInfo* bd_fs_ntfs_get_info (const gchar *device, GError **error);
 
+/**
+ * bd_fs_f2fs_mkfs:
+ * @device: the device to create a new f2fs fs on
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the creation (right now
+ *                                                 passed to the 'mkfs.f2fs' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether a new f2fs fs was successfully created on @device or not
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_MKFS
+ */
+gboolean bd_fs_f2fs_mkfs (const gchar *device, const BDExtraArg **extra, GError **error);
+
+/**
+ * bd_fs_f2fs_wipe:
+ * @device: the device to wipe an f2fs signature from
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an f2fs signature was successfully wiped from the @device or
+ *          not
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_WIPE
+ */
+gboolean bd_fs_f2fs_wipe (const gchar *device, GError **error);
+
+/**
+ * bd_fs_f2fs_check:
+ * @device: the device containing the file system to check
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the repair (right now
+ *                                                 passed to the 'fsck.f2fs' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an f2fs file system on the @device is clean or not
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_CHECK
+ */
+gboolean bd_fs_f2fs_check (const gchar *device, const BDExtraArg **extra, GError **error);
+
+/**
+ * bd_fs_f2fs_repair:
+ * @device: the device containing the file system to repair
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the repair (right now
+ *                                                 passed to the 'fsck.f2fs' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an f2fs file system on the @device was successfully repaired
+ *          (if needed) or not (error is set in that case)
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_REPAIR
+ */
+gboolean bd_fs_f2fs_repair (const gchar *device, const BDExtraArg **extra, GError **error);
+
+/**
+ * bd_fs_f2fs_get_info:
+ * @device: the device containing the file system to get info for
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: (transfer full): information about the file system on @device or
+ *                           %NULL in case of error
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_QUERY
+ */
+BDFSF2FSInfo* bd_fs_f2fs_get_info (const gchar *device, GError **error);
+
+/**
+ * bd_fs_f2fs_resize:
+ * @device: the device containing the file system to resize
+ * @new_size: new requested size for the file system *in file system sectors* (see bd_fs_f2fs_get_info())
+ *            (if 0, the file system is adapted to the underlying block device)
+ * @safe: whether to perform safe resize or not (does not resize metadata)
```

Comment:
Why would somebody want to do an unsafe resize?

Reply:
So I see it is required for shrinking. I think it should just happen automatically. Or is there a use case where it would make sense to not use it when needed?

Reply:
I think better question is if there is a use case for "safe" resize when it's not needed (for growing). And the answer is "I don't know". I think the best solution is to give user full control here, the generic resize function is adding the safe option for shrinking which should be ok for most users IMHO.

Reply:
Works for me.